### PR TITLE
Fix decompression error on HEAD request

### PR DESCRIPTION
**Pull request by dergecko**
Fixing a bug mentioned in #5102.

HEAD requests will fail if the decompression is set in the result header as there is no data to decompress.
Hope the test is at the right place if not let me know how to improve!

### CODE REVIEW

File: lib/adapters/http.js [hunk 1]
```diff
@@ -380,9 +380,9 @@ export default function httpAdapter(config) {
 
       // if decompress disabled we should not decompress
       if (config.decompress !== false) {
-        // if no content, but headers still say that it is encoded,
+        // if no content or HEAD request, but headers still say that it is encoded,
         // remove the header not confuse downstream operations
-        if (data && data.length === 0 && res.headers['content-encoding']) {
+        if ((method === 'HEAD' || data && data.length === 0) && res.headers['content-encoding']) {
```
Comment (lejahmie) [inline lib/adapters/http.js:385]: I assume that `data` is `undefined` in the scenario? Would it not be better to just check that then?

```
(typeof data === 'undefined' || data && data.length === 0)
```

Feels safer than making specific checks against method?

Comment (dergecko) [reply lib/adapters/http.js:385]: The problem is this `data` is always undefined for `GET` requests and would also skip the all decompression on those.
As far as I could comprehend, `data` is only set before hand if it is a `POST` request.
The response stream is handled after this `data` variable is checked. If you would include this `undefined` check the unit tests 
 - should support gunzip error handling
 - should support transparent gunzip
will fail as they would not decompress the response anymore.

Comment (dergecko) [reply lib/adapters/http.js:386]: I close this issue, if you have further questions just write a comment again.

**Comment by dergecko** [comment]
Do you need any further information for this PR to progress?
If so just ping me here.

**Comment by gcampionpae** [comment]
Is there anything else needed here to get this bug fixed?

**Comment by jasonsaayman** [comment]
@dergecko please can you resolve the merge conflict? Thanks

**Comment by DigitalBrainJS** [comment]
I'm going to close this as it seems the issue [has been fixed](https://github.com/axios/axios/blob/726f1c8e00cffa0461a8813a9bdcb8f8b9d762cf/lib/adapters/http.js#L468) with another PR. Thanks for your contribution!

**Pull request closed without merging**